lib/diagnostic.rb

```diff
@@ -7,36 +7,36 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max_by(&:length)
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.inject(0, :+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.values_at(* numbers.each_index.select {|i| i.odd? })
   end
```

Comment:
For Qs 3 and 4, I'm not sure where you got values_at from, but that's a method on Hashes. What is the datatype of numbers? It is a list.

Here's 1 solution:

```
 def some_odd(numbers)
    numbers.any?(&:odd?)
  end
```